File: week15/p_43162.py

computers = [[1, 1, 0], [1, 1, 1], [0, 1, 1]]	
n = 3
visited = [0]*n
def solution(n):
    count = 0
    for i in range(n):
        if visited[i] != 1:
            bfs(i)
            count += 1
    return count


def bfs(i):
    q = [i]
    visited[i] = 1
    while q:
        t = q.pop(0)
        for i in range(len(computers[t])):
            if computers[t][i] == 1 and visited[i] == 0:
                q.append(i)
                visited[i] = 1
        
print(island(n))

File: week20/p_43162.js
function solution(n, computers) {
    var answer = 0;
    let visited = [false];
    
    function dfs(i){
        visited[i] =[true];
        
        for (let j = 0; j<computers[i].length;j++){
            if(computers[i][j]==1 && !visited[j]){
                dfs(j)
            }
        }
           
    }
    
    for(let i=0; i<n ; i++){
        if(!visited[i]){
            dfs(i)
            answer += 1
        }
    }
    return answer;
}